pkg/api/graph/graph.go

```diff
@@ -114,6 +111,23 @@ type Interface interface {
 	Edges() []graph.Edge
 }
 
+type Namer interface {
+	ResourceName(obj interface{}) string
+}
+
+type namer struct{}
+
+var DefaultNamer Namer = namer{}
+
+func (namer) ResourceName(obj interface{}) string {
+	switch t := obj.(type) {
+	case uniqueNamer:
+		return t.UniqueName()
```

Comment:
The previous intent was "resourceTypeShortName/resourceName".  UniqueName has different constraints.  As I recall, I have it doing pipes and namespaces.

I really want to be able to copy/paste the output to `kubectl` and I think that this change makes it impossible to do that in the common case.